### ISSUE TITLE
SC2034 hit even though variable is used at end of function

### ISSUE THREAD

**Issue by mmlb** [koalaman/shellcheck#1213]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

function foo() {
        local bar=$1
        shift
        local -A aarray=(
                [a]=6
        )
        ((5 > aarray[bar]))
}
```

#### Here's what shellcheck currently says:
```
Line 4	SC2034: bar appears unused. Verify it or export it.
```
#### Here's what I wanted or expected to see:

No errors since the variable is used in the math test expression at the end of the function.

**Comment by matthewpersico** [comment]
The statement `((5 > aarray[bar]))`, is not doing what you think it is.
I created another script to demonstrate. See this and its output:

```
#!/bin/bash
declare bar=a 
declare -A aarray=([a]=6)
echo "${aarray[a]}"
if ((5 > aarray[bar]))
then
echo "true to 5 > ${aarray[$bar]}"
else 
echo "false to 5 > ${aarray[$bar]}"
fi
if ((5 > aarray[$bar]))
then
echo "true to 5 > ${aarray[$bar]}"
else
echo "false to 5 > ${aarray[$bar]}"
fi
```
```
./scissue.1213
6
true to 5 > 6
false to 5 > 6
```
I think that the statement `aarray[bar]` doesn't work in (()). You can drop the $ on the associative array, but not the index. And no, I have no idea why formatting isn't working.

**Comment by mmlb** [comment]
:man_facepalming: makes total sense!
